src/ts/proxy.ts

```diff
@@ -0,0 +1,45 @@
+// defined in https://eips.ethereum.org/EIPS/eip-1967
+
+import { BigNumber } from "ethers";
+import { ethers } from "hardhat";
+
+// The proxy contract used by hardhat-deploy implements EIP-1967 (Standard Proxy
+// Storage Slot). See <https://eips.ethereum.org/EIPS/eip-1967>.
+function slot(string: string) {
+  return ethers.utils.defaultAbiCoder.encode(
+    ["bytes32"],
+    [BigNumber.from(ethers.utils.id(string)).sub(1)],
+  );
+}
+const IMPLEMENTATION_STORAGE_SLOT = slot("eip1967.proxy.implementation");
+const OWNER_STORAGE_SLOT = slot("eip1967.proxy.admin");
+
+/**
+ * Returns the address of the implementation of an EIP-1967-compatible proxy
+ * from its address.
+ *
+ * @param proxy Address of the proxy contract.
+ * @returns The address of the contract storing the proxy implementation.
+ */
+export async function implementationAddress(proxy: string): Promise<string> {
+  const [implementation] = await ethers.utils.defaultAbiCoder.decode(
+    ["address"],
+    await ethers.provider.getStorageAt(proxy, IMPLEMENTATION_STORAGE_SLOT),
+  );
+  return implementation;
+}
+
+/**
+ * Returns the address of the imple mentation of an EIP-1967-compatible proxy
+ * from its address.
+ *
+ * @param proxy Address of the proxy contract.
+ * @returns The address of the administrator of the proxy.
+ */
+export async function ownerAddress(proxy: string): Promise<string> {
```

Comment:
uber-naming-nit: Maybe we should keep the same naming as defined in EIP-1967
```suggestion
export async function adminAddress(proxy: string): Promise<string> {
```

I'm OK with either, just a minor suggestion.

Reply:
The problem with this is that the proxy implementation of `hardhat-deploy` uses `owner` in lieu of `admin`, the latter being used in the standard. Otherwise I'd have liked everywhere in this PR `s/owner/admin` and `s/manager/owner`.

Reply:
Ok, so based on @fedgiac's comment, should I do nothing or make this change?

Reply:
Maybe it makes sense to match the library's naming so it doesn't cause confusion as @fedgiac suggested. I mentioned, I'm really OK with either.

Reply:
Ok, will leave this for now. Its not a smart contract issue so it can always be updated after the audit/deployment etc...